### DISCUSSION TITLE
Adding Sub Region Column to TSML UI list view

### DISCUSSION THREAD

**Discussion by steve970** [code4recovery/12-step-meeting-list#1567] [Q&A] [answered]
Hello,

I am in the process of implementing this plugin to display international meetings. It would be really helpful to use countries as regions and cities as sub-regions and be able to display both on the list view. Can you help?

**Comment by joshreisner** [comment]
hi @steve970 we wouldn't add sub region as a column per se to TSML UI, since meetings can have n number of regions in their hierarchy. we might be able to do something like this, if that is appealing? we would need to think this through, it might not be helpful in all cases

<img width="1219" alt="Screenshot 2024-11-15 at 1 15 28 PM" src="https://github.com/user-attachments/assets/712afb25-24fc-4a85-975c-31b3676ff91a">

**Reply by joshreisner** [reply, marked as answer]
what that would mean for you: your meetings would have region column values like `Germany > Berlin` or `USA > Illinois > Chicago`

**Comment by steve970** [comment]
Really what I am trying to find is a solution for both the Region dropdown menu and then what the list view shows after. If I am looking at this site for an example, I am guessing this is the Legacy UI. What dictates what shows up in the Region column on the list view? I see some are the higher level and some are the regional?

[https://bayareasaa.org/meetings/?tsml-day=any&tsml-region=north-bay]( https://bayareasaa.org/meetings/?tsml-day=any&tsml-region=north-bay)

**Reply by SIA-WEBSUP** [reply]
@steve970 , I think you are correct this site uses the Legacy UI and therefore has collapsible elements in the regions dropdown.

I'm almost certain that currently the **Region** column displays the terminal node in the 'region/subregion' tree. These trees are rarely very deep, so Josh was asking if displaying the entire tree would satisfy your needs.
     i.e. `Top-Region > Sub-Region > Sub-Sub-Region `

**Reply by steve970** [reply]
Please see the response I put in this thread. I think that displaying just the Region would suffice if the menu was collapsable when filtering by region.  Thanks!

https://github.com/code4recovery/tsml-ui/issues/411
